### ISSUE TITLE
Extension is Not working 

### ISSUE THREAD

**Issue by nocivix** [krisztianb/seekingalpha-paywall-remover#11]
Hi

Fyi , Probably they changjed something

try : https://seekingalpha.com/news/4268553-prestige-wealth-acquires-innosphere-tech

Not working 

Many THX

**Comment by lattaai5** [comment]
Hello, I tried to fix the issue

This is what I did:

Modified the paywall-remover.js file to address changes in the Seeking Alpha website structure. Updated selectors and added new logic to handle different page layouts, including the news pages.

I also created pull request: https://github.com/krisztianb/seekingalpha-paywall-remover/pull/12


> [!CAUTION]
> Disclaimer: This fix was created by Latta AI and you should never merge before you check the correctness of generated code!
>
> The fix provided by Latta AI might not be complete and it can serve as an inspiration.

---
This bug was fixed for free by Latta AI - https://latta.ai/ourmission

If you no longer want Latta AI to attempt fixing issues on your repository, you can block this account.

**Comment by krisztianb** [comment]
Hi. Indeed it doesn't seem to be working anymore. Thank you for reporting.

**Comment by krisztianb** [comment]
So they finally changed their site so that they no longer load the full article unless you are registered. This means the extension is finally forever broken. I'm going to lock this repository and remove the extension from the Firefox extension hub. Have a nice day guys.

**Comment by krisztianb** [comment]
I'm totally confused. I was debugging this and couldn't figure out a solution as the full article was no longer loaded. After deleting all cookies in the browser and reinstalling the extension it suddenly started working again.

This is the page you mentioned:

![image](https://github.com/user-attachments/assets/4d7d94e2-30f4-4e38-afee-3eff8fdc7d95)

**Comment by nocivix** [comment]
THX, They got scared :)

**Comment by Combowombo** [comment]
yeah i noticed that seekingalpha had a new banner at the bottom and i think i'm on the version of this extension from september so it hasn't worked for me the past week. Glad someone posted a comment, I'll delete my cookies and try again. 

Also, want to give a thanks to the maintainer as this extension has greatly benefitted my ability to do stock research

**Comment by krisztianb** [comment]
Can anybody confirm that it is working for them too?

**Comment by almadimate** [comment]
> Can anybody confirm that it is working for them too?

Yes, it works for me too!
In addition, it's not needed to refresh the page several times to read an article. :)

**Comment by nocivix** [comment]
it is strange , there is Not logic

Sometime if I cancel cache is ok

othertimes NOT , i have ALSP to close and restart Firefox

Seems YES the proble is in the cache

BUT by example still not working on few pages like this, I suppose is RND issues

https://seekingalpha.com/news/4273709-japanese-govt-to-provide-65b-in-subsidies-for-chip-industry-report

**Comment by nocivix** [comment]
Seems not working 
https://seekingalpha.com/news/4274512-chips-act-appears-safe-as-white-house-changes-hands-needham

**Comment by nocivix** [comment]
Not working 
https://seekingalpha.com/article/4735965-microsoft-stock-fundamentals-resilient-unremarkable-growth-rating-downgrade

**Comment by krisztianb** [comment]
@nocivix both pages work if I'm logged in.

If I'm logged out the news page is blocked by the paywall and the extension is not able to remove it. The article doesn't even load if I'm logged out (even when the extension is disabled) I get this:

![Screenshot from 2024-11-11 19-59-48](https://github.com/user-attachments/assets/d48679e2-596f-4c2a-aadb-c6869d3ba283)

Are you guys logged in or not?

**Comment by Combowombo** [comment]
i'm logged in and its not working. Although for me i've actually taken the code and put it in a tampermonkey script rather than use the extension (since i'm using qutebrowser).

**Comment by nocivix** [comment]
I confirm is not working in Firefox

**Comment by nocivix** [comment]
![image](https://github.com/user-attachments/assets/f0817161-e11c-4229-b76d-437b2f49647d)

**Comment by nocivix** [comment]
My page is this, the usual one

**Comment by nocivix** [comment]
https://seekingalpha.com/news/4275270-openai-applies-new-techniques-to-improve-upcoming-model-report

**Comment by nocivix** [comment]
What is strange now I cannot see ARTCLE but I can see the comments
![image](https://github.com/user-attachments/assets/402ee25c-57b7-47b4-93a4-e4342417902c)

**Comment by Blackdog702024** [comment]
Hi all, any update on how the fix is going? still not working for me.
https://seekingalpha.com/news/4284594-cpi-read-food-prices-ran-below-the-overall-inflation-rate-in-october?mailingid=37457815&messageid=2900&serial=37457815.43128&source=email_2900

**Comment by krisztianb** [comment]
Hi. I will probably have time at the weekend to look into it. However it seemed to me that they modified the site in a way that the full article is no longer pre-loaded. Which would mean that the extension cannot be adjusted to work again.

But the link you posted works for me. Also sometimes logging out and back in removes the paywall. I don't fully understand how the SA site works. Sometimes it seems to randomly work again. Like @nocivix said there seems to be no logic.

**Comment by Rincemac** [comment]
I suspect they are doing some tech A/B Testing, just as a possible source of the "randomness" experienced while loading (supported by the fact of different users getting different results).

**Comment by isocolon** [comment]
I agree. Started to notice this 1-2 months back. For some time, the paywall got moved to server-side, then it was back to normal again. Now it does not seem to be working again.

**Comment by krisztianb** [comment]
I analyzed this a bit further and see the following happening on my side:

* When one is not logged in the full article is no longer loaded temporarily before it gets hidden by the paywall. Therefore the extension doesn't do anything because it doesn't see the fully loaded article. => I think this can't be fixed.
* When you are logged in I get no pay wall at all and see the full article upfront. Which means that the extension is also not triggered. => This is great. Makes the extension obsolete.

Maybe they broke their own pay wall? There is nothing to do here. Uninstall the extension. Log in and enjoy SA without the paywall. :smile:

**Comment by almadimate** [comment]
I'm logged in and I've got the paywall: https://seekingalpha.com/article/4733735-pepsico-healthy-beverage-dtc-initiatives-fuel-long-term-growth

Maybe they gave you a subscription for free in order to stop the development. :D

**Comment by Begiz88** [comment]
Hi, unfortunately for me he extension is not working anymore.
I've tried your solutions but i still cannot read the articles.
Could you help me? Thank you very much

**Comment by Blackdog702024** [comment]
As above i uninstalled extension but alas still get blocked.
https://seekingalpha.com/article/4738222-bristol-myers-squibb-buy-this-bargain-stock-now?mailingid=37511230&messageid=2800&serial=37511230.12734&source=email_2800

**Comment by krisztianb** [comment]
> I'm logged in and I've got the paywall: https://seekingalpha.com/article/4733735-pepsico-healthy-beverage-dtc-initiatives-fuel-long-term-growth
> 
> Maybe they gave you a subscription for free in order to stop the development. :D

I will create a new account and check it out. 👍

**Comment by rodrmp** [comment]
I suspect there is a limit of number of articles you can read for free in a month? you can try opening loads of them until it paywalls you again. 

This extension definitely stopped working both logged in or not.  Some articles can be read using archived pages (web.archive, waybackmachine etc) but not comments, which is the only interesting thing in seekingalpha

**Comment by Blackdog702024** [comment]
Rodrmp,i use a web proxy to read the comments,used to load the whole article but seeking alpha changed something a couple months ago so alas only get the first part now,please be aware if you use web proxys some will be blocked by your anti-virus software.

**Comment by krisztianb** [comment]
Now I also get the paywall even if I'm logged in. The page no longer preloads the full article content. So the extension can't be updated to work again. They finally fixed their site so that it only loads the full article once your subscription is verified. 💩

**Comment by Norjak** [comment]
> I analyzed this a bit further and see the following happening on my side:
> 
>     * When one is not logged in the full article is no longer loaded temporarily before it gets hidden by the paywall. Therefore the extension doesn't do anything because it doesn't see the fully loaded article. => I think this can't be fixed.
> 
>     * When you are logged in I get no pay wall at all and see the full article upfront. Which means that the extension is also not triggered. => This is great. Makes the extension obsolete.
> 
> 
> Maybe they broke their own pay wall? There is nothing to do here. Uninstall the extension. Log in and enjoy SA without the paywall. 😄

It looks like they temporarily disabled it during their Black Friday promotions last week. The paywall is back today, along with an offer for 20% off an annual Premium membership.

**Comment by Combowombo** [comment]
awww that's too bad, it was good while it lasted. Its too bad there's no alternative to seeking alpha articles, value investors club just doesn't have enough write ups
